azure-kusto-ingest/src/resource_manager/ingest_client_resources.rs

```diff
@@ -0,0 +1,139 @@
+use std::sync::Arc;
+
+use crate::client_options::QueuedIngestClientOptions;
+
+use super::{
+    cache::{Cached, ThreadSafeCachedValue},
+    resource_uri::{ClientFromResourceUri, ResourceUri},
+    utils::get_column_index,
+    RESOURCE_REFRESH_PERIOD,
+};
+use anyhow::Result;
+use azure_core::ClientOptions;
+use azure_kusto_data::{models::TableV1, prelude::KustoClient};
+use azure_storage_blobs::prelude::ContainerClient;
+use azure_storage_queues::QueueClient;
+use tokio::sync::RwLock;
```

Comment:
We rather not take a dep on tokio so we can run on any event loop. Is there not a "generic" async rwlock?

Reply:
There is a `RwLock` in the standard library which we could change to:
https://doc.rust-lang.org/std/sync/struct.RwLock.html. 

The following may be relevant and likely applies to the RwLock:
https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use

It's worth noting this issue around tokio usage in the Rust SDK: https://github.com/Azure/azure-sdk-for-rust/issues/1469

Reply:
Ended up going for the RwLock that the Azure SDK uses which is [async-lock](https://crates.io/crates/async-lock)

Reply:
awesome!